src/main/java/uk/gov/hmcts/reform/jps/services/refdata/LocationService.java

```diff
@@ -18,6 +18,33 @@
 public class LocationService {
     private final LocationServiceClient regionServiceClient;
 
+    public Optional<CourtVenue> getCourtVenue(String hmctsServiceCode, String epimmsId) {
+        return getCourtVenue(
+            getLocationApiResponse(hmctsServiceCode),
+            epimmsId,
+            (court, epimsId) -> court.getEpimmsId().equals(epimsId)
+        );
+    }
+
+    private Optional<CourtVenue> getCourtVenue(LocationApiResponse serviceCourtInfo,
+                                               String value,
+                                               BiPredicate<CourtVenue, String> predicate) {
+        if (null == serviceCourtInfo) {
```

Comment:
This could be changed to 
return Optional.ofNullable(serviceCourtInfo.getCourtVenues()).stream()
            .filter(courtVenue -> predicate.test(courtVenue, value))
            .findAny();

Reply:
Tried that but it doesn't work as is